src/utils/relayer.ts

```diff
@@ -0,0 +1,110 @@
+import { Redis } from 'ioredis'
+import { HttpErrors } from '@loopback/rest'
+import { Node } from '@pokt-network/pocket-js'
+
+import { BlockchainsRepository } from '../repositories'
+import { SyncCheckOptions } from '../services/sync-checker'
+import { DEFAULT_BLOCK_LOGS_LIMIT } from '../utils/constants'
+import { BlockchainDetails } from './types'
+
+// Fetch node client type if Ethereum based
+export async function fetchClientTypeLog(
+  redis: Redis,
+  blockchainID: string,
+  id: string | undefined
+): Promise<string | null> {
+  const clientTypeLog = await redis.get(`${blockchainID}-${id}-clientType`)
+
+  return clientTypeLog
+}
+
+export function isCheckPromiseResolved(promise: PromiseSettledResult<Node[]>): boolean {
+  return promise.status === 'fulfilled' && promise.value !== undefined && promise.value.length > 0
+}
+
+export function filterCheckedNodes(syncCheckNodes: Node[], chainCheckedNodes: Node[]): Node[] {
+  // Filters out nodes that passed both checks.
+  const nodes = syncCheckNodes.filter((syncCheckNode) =>
+    chainCheckedNodes.some((chainCheckedNode) => syncCheckNode.publicKey === chainCheckedNode.publicKey)
+  )
+
+  return nodes
+}
+
+// Load requested blockchain by parsing the URL
+export async function loadBlockchain(
+  host: string,
+  redis: Redis,
+  blockchainsRepository: BlockchainsRepository,
+  defaultLogLimitBlocks: number
+): Promise<BlockchainDetails> {
+  // Load the requested blockchain
+  const cachedBlockchains = await redis.get('blockchains')
+  let blockchains
+
+  if (!cachedBlockchains) {
+    blockchains = await blockchainsRepository.find()
+    await redis.set('blockchains', JSON.stringify(blockchains), 'EX', 60)
+  } else {
+    blockchains = JSON.parse(cachedBlockchains)
+  }
+
+  // Split off the first part of the request's host and check for matches
+  const [blockchainRequest] = host.split('.')
+
+  const [blockchainFilter] = blockchains.filter(
+    (b: { blockchain: string }) => b.blockchain.toLowerCase() === blockchainRequest.toLowerCase()
+  )
+
+  if (blockchainFilter) {
```

Comment:
here comes I with my pet peeves, I'm a fan of having the main flow outside any block, that means the expected return value should not be anywhere else but at the end of the file, so when I read code I know that everything outside blocks is the "desired" behavior. So why not better change the `if` to check if not there and throw on that case and continue the flow normally on the main block.